### PR TITLE
[Dygraph] add FCOS-DCN

### PR DESCRIPTION
**Pull request by nemonameless**
1. add DCN on ResNet
2. add DCN on FCOS head
3. add FCOS multiscale training

### CODE REVIEW

File: dygraph/ppdet/modeling/backbones/resnet.py [hunk 1]
```diff
@@ -41,22 +42,36 @@ def __init__(self,
...
-            in_channels=ch_in,
-            out_channels=ch_out,
-            kernel_size=filter_size,
-            stride=stride,
-            padding=(filter_size - 1) // 2,
-            groups=1,
-            weight_attr=ParamAttr(
-                learning_rate=lr, name=name + "_weights"),
-            bias_attr=False)
+        if dcn_v2 is False:
```
Comment (heavengate) [inline dygraph/ppdet/modeling/backbones/resnet.py:52]: if not dcn_v2

File: dygraph/ppdet/modeling/layers.py [hunk 2]
```diff
@@ -62,19 +134,38 @@ def __init__(self,
...
-            stride=stride,
-            padding=(filter_size - 1) // 2,
-            groups=1,
-            weight_attr=ParamAttr(
-                name=name + "_weight",
-                initializer=Normal(
-                    mean=0., std=0.01),
-                learning_rate=1.),
-            bias_attr=bias_attr)
+        if use_dcn == False:
```
Comment (heavengate) [inline dygraph/ppdet/modeling/layers.py:137]: if not use_dcn

File: dygraph/ppdet/modeling/layers.py [hunk 3]
```diff
@@ -62,19 +134,38 @@ def __init__(self,
...
+                stride=stride,
+                padding=(filter_size - 1) // 2,
+                groups=1,
+                weight_attr=ParamAttr(
+                    name=name + "_weight",
+                    initializer=Normal(
+                        mean=0., std=0.01),
+                    learning_rate=1.),
+                bias_attr=True,
+                lr_scale=2.,  #
```
Comment (heavengate) [inline dygraph/ppdet/modeling/layers.py:166]: delete  #

File: dygraph/ppdet/modeling/layers.py [hunk 4]
```diff
@@ -38,6 +39,77 @@ def _to_list(l):
...
+                name='{}._conv_offset.weight'.format(name)),
+            bias_attr=offset_bias_attr)
+
+        if bias_attr:
+            # in FCOS-DCN head, specifically need learning_rate and regularizer
+            dcn_bias_attr = ParamAttr(
+                name=name + "_bias",
+                initializer=Constant(value=0),
+                regularizer=L2Decay(0.),  #
+                learning_rate=2.)  #
```
Comment (yghstill) [inline dygraph/ppdet/modeling/layers.py:87]: delete #

**Review by yghstill — APPROVED** [review]
LGTM

**Pull request merged**